src/main/features/directions-questionnaire/forms/models/availability.ts

```diff
@@ -0,0 +1,55 @@
+import { IsDefined } from '@hmcts/class-validator'
+import { ValidationErrors as GlobalValidationErrors } from 'forms/validation/validationErrors'
+import { YesNoOption } from 'models/yesNoOption'
+import { LocalDate } from 'forms/models/localDate'
+import * as moment from 'moment'
+import { IsValidLocalDate } from '@hmcts/cmc-validators'
+import { IsFutureDate } from 'forms/validation/validators/dateFutureConstraint'
+
+export class ValidationErrors {
+  static readonly AT_LEAST_ONE_DATE: string = 'Select at least one date or choose No'
+  static readonly DATE_NOT_VALID: string = 'Please enter a valid date'
+  static readonly FUTURE_DATE: string = 'Enter a date in the future'
+}
+
+export class Availability {
+
+  @IsDefined({ message: GlobalValidationErrors.YES_NO_REQUIRED })
+  hasUnavailableDates?: YesNoOption
+
+  unavailableDates?: LocalDate[]
```

Comment:
No validation here. With the JS page, only valid dates should be possible; with the non-JS page validation is done via the `newDate`. Not sure this is appropriate nevertheless.

Reply:
Revisit - changed a lot since I removed the validation here

Reply:
`@ValidateNested()` was misunderstood. It applies the validations found on the field's type's fields (to each item of the array if it's an array) - _not_ the validations listed at this level.